File: Submission-Java/app/src/main/java/com/ahmadabuhasan/appgithubuser/networking/ApiService.java
package com.ahmadabuhasan.appgithubuser.networking;

import com.ahmadabuhasan.appgithubuser.model.Follow;
import com.ahmadabuhasan.appgithubuser.model.Search;
import com.ahmadabuhasan.appgithubuser.model.User;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @GET("users/{username}")
    Call<User> detailUser(
            @Path("username") String username);

    @GET("/search/users")
    Call<Search> searchUser(
            @Query("q") String username);

    @GET("/users/{username}/followers")
    Call<ArrayList<Follow>> followersUser(
            @Path("username") String username);

    @GET("users/{username}/following")
    Call<ArrayList<Follow>> followingUser(
            @Path("username") String username);
}

File: Submission-Kotlin/app/src/main/java/com/ahmadabuhasan/appgithubuser/api/ApiService.kt
package com.ahmadabuhasan.appgithubuser.api

import com.ahmadabuhasan.appgithubuser.model.ResponseDetailUser
import com.ahmadabuhasan.appgithubuser.model.ResponseFollow
import com.ahmadabuhasan.appgithubuser.model.ResponseSearch
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {

    @GET("search/users")
    fun search(
        @Query("q") username: String
    ): Call<ResponseSearch>

    @GET("users/{username}")
    fun detailUser(
        @Path("username") username: String
    ): Call<ResponseDetailUser>

    @GET("users/{username}/followers")
    fun followers(
        @Path("username") username: String
    ): Call<ArrayList<ResponseFollow>>

    @GET("users/{username}/following")
    fun following(
        @Path("username") username: String
    ): Call<ArrayList<ResponseFollow>>
}